### PR TITLE
Added any? and all? methods

### PR DESCRIPTION
**Pull request by ndbroadbent**
See: #198

I think these could be very useful.

### CODE REVIEW

File: lib/parallel.rb [hunk 1]
```diff
@@ -215,6 +215,14 @@ def each(array, options={}, &block)
       array
     end
 
+    def any?(array, options={}, &block)
+      map(array, options.merge(:break_condition => :any), &block || -> (v) { v })
+    end
+
+    def all?(array, options={}, &block)
+      map(array, options.merge(:break_condition => :all), &block || -> (v) { v })
```
Comment (grosser) [inline lib/parallel.rb]: return `!all(*args)` here ?

**Comment by grosser** [comment]
lots of code for a simple change ... I was hoping for something that did not require changing the internals like

```
def any?(*args, &block)
  !each(*args) { raise Parallel::Kill if block.call }
end
```

**Comment by ndbroadbent** [comment]
Thanks for the feedback, I agree this is not very elegant. I'll try to trim it down to something more reasonable

**Comment by ndbroadbent** [comment]
Updated!

Haha you're right, this is WAY better. I just needed to fix that `work_direct` issue in #201

File: Readme.md [hunk 2]
```diff
@@ -37,6 +37,18 @@ items = [1,2,3]
 Parallel.each( -> { items.pop || Parallel::Stop }) { |number| ... }
 ```
 
+You can also call `any?` or `all?`, which work the same way as `Array#any?` and `Array#all?`.
+
+```Ruby
+items = [1,2,3,4,5,6,7]
+Parallel.any?(items) { |number| number == 4 }
```
Comment (grosser) [inline Readme.md]: save a line by inlining the `items=` ?

File: lib/parallel.rb [hunk 3]
```diff
@@ -215,6 +215,20 @@ def each(array, options={}, &block)
       array
     end
 
+    def any?(array, options={}, &block)
+      block ||= -> (v) { v }
+      !map(array, options.merge(:preserve_results => false)) do |*args|
```
Comment (grosser) [inline lib/parallel.rb]: can use `each` ?

File: lib/parallel.rb [hunk 4]
```diff
@@ -215,6 +215,20 @@ def each(array, options={}, &block)
...
+        raise Parallel::Kill if block.call(*args)
+      end
+    end
+
+    def all?(array, options={}, &block)
+      block ||= -> (v) { v }
```
Comment (grosser) [inline lib/parallel.rb]: better raise when there is no block ?

**Comment by grosser** [comment]
lookin good!

File: lib/parallel.rb (hunk 3 above)
Comment (ndbroadbent) [inline lib/parallel.rb]: `#each` returns the original array, so I couldn't do that, I need the result of the `map` call

File: Readme.md (hunk 2 above)
Comment (ndbroadbent) [inline Readme.md]: 👍

File: lib/parallel.rb (hunk 4 above)
Comment (ndbroadbent) [inline lib/parallel.rb]: Ah, the implicit block is the default behavior in Ruby: http://apidock.com/ruby/Enumerable/any%3F

So you can call:

```
[ nil, true, 99 ].any?                            #=> true
```

**Comment by grosser** [comment]
yeah, but kind of weird/useless when you call parallel without a block ?

On Fri, Mar 10, 2017 at 12:34 PM, Nathan Broadbent <notifications@github.com

**Comment by grosser** [comment]
I guess that's another bug then ... .each should return nil on break:

```
[1,2,3].each{ break }
=> nil
```

... but can be it's own PR if it's not trivial to fix

On Fri, Mar 10, 2017 at 12:32 PM, Nathan Broadbent <notifications@github.com

**Comment by ndbroadbent** [comment]
I pushed a fix for the each issue, so now I'm using each. Super clean!

**Comment by grosser** [comment]
❤️  waiting for green :)

**Comment by grosser** [comment]
1.11.0

**Comment by ndbroadbent** [comment]
Awesome, thanks very much!

**Pull request merged**